server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -148,6 +149,7 @@ private RolloverResult rolloverDataStream(ClusterState currentState, IndexAbstra
         final DataStream ds = dataStream.getDataStream();
         final IndexMetadata originalWriteIndex = dataStream.getWriteIndex();
         final String newWriteIndexName = DataStream.getDefaultBackingIndexName(ds.getName(), ds.getGeneration() + 1);
+        ds.rollover(new Index(newWriteIndexName, "uuid")); // just for validation
```

Comment:
I think that this is what happens in this `rolloverDataStream(...)` method. Anything before the if statement is for validating purposes and after the if statement is for actually performing the rollover (the `DataStream#rollover(...)` method is then also invoked). I just added the `DataStream#rollover(...)` invocation here, so that validation that is in this method is also performed when just validating the rollover.